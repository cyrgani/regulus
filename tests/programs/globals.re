def(f, _(
    global(G),
)),

def(e, _(
    =(G, 2),
)),

__builtin_file_catch_assert_eq(G),
e(),
__builtin_file_catch_assert_eq(G),
f(),
__builtin_file_catch_assert_eq(G),
e(),
assert_eq(G, 2),

import(globals_helper),
__builtin_file_catch_assert_eq(use_global()),
=(GLOBALS_HELPER, 5),
use_global(),
square_glob(),
assert_eq(GLOBALS_HELPER, 25),
