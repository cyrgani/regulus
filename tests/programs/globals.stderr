NameError: No variable named `G` found!
at programs/globals.re:9:23
at programs/globals.re:9:22
NameError: No variable named `G` found!
at programs/globals.re:11:23
at programs/globals.re:11:22
at programs/globals.re:9:22
AssertionError: Equality assertion failed! lhs: `1`, rhs: `5`!
at <stl:assertions>:14:10
at <stl:assertions>:11:32
at programs/globals_helper.re:3:26
at programs/globals.re:18:33
at programs/globals.re:18:22
at programs/globals.re:11:22
at programs/globals.re:9:22
